tests/integration/targets/azure_rm_keyvault/tasks/main.yml

```diff
@@ -283,3 +304,270 @@
+    that:
+      - output.changed == false
+
+- name: Update existing HSM (add tags)
+  azure_rm_keyvault:
+    resource_group: "{{ resource_group }}"
+    hsm_name: "hsm{{ rpfx }}"
+    tenant_id: "{{ tenant_id }}"
+    administrators:
+      - "{{ object_id }}"
+    identity:
+      type: UserAssigned
+      user_assigned_identity: "{{ managed_identity_ids[0] }}"
+    soft_delete_retention_in_days: 7
+    sku:
+      name: Standard_B1
+      family: B
+    tags:
+      aaa: bbb
+  register: output
+
+- name: Assert the state has changed
+  ansible.builtin.assert:
+    that:
+      - output.changed == true
+
+- name: Get hsm facts
+  azure_rm_keyvault_info:
+    resource_group: "{{ resource_group }}"
+    hsm_name: "hsm{{ rpfx }}"
+  register: facts
+
+- name: Assert the facts are properly set
+  ansible.builtin.assert:
+    that:
+      - facts['hsms'] | length == 1
+      - facts['hsms'][0]['hsm_uri'] != None
+      - facts['hsms'][0]['name'] != None
+      - facts['hsms'][0]['sku'] != None
+      - facts['hsms'][0]['id'] != None
+      - facts['hsms'][0]['enable_soft_delete'] == true
+      - facts['hsms'][0]['soft_delete_retention_in_days'] == 7
+
+- name: Download the security domain file
+  azure_rm_keyvaultsecuritydomain:
+    keyvault_uri: https://hsm{{ rpfx }}.managedhsm.azure.net
+    sd_quorum: 2
+    sd_wrapping_keys:
+      - "{{ lookup('file', 'cert_0.cer') }}"
+      - "{{ lookup('file', 'cert_1.cer') }}"
+      - "{{ lookup('file', 'cert_2.cer') }}"
+    action: download
+  register: output
+
+- name: Save the security domain file
+  ansible.builtin.debug:
+    var: output["security_domain"]
+
+#
+# azure_rm_hsmkey tests
+#
+
+- name: Create a hsm key
+  block:
+    - name: Create a hsm key
+      azure_rm_keyvaultkey:
+        keyvault_uri: https://hsm{{ rpfx }}.managedhsm.azure.net
+        key_name: testkey
+        tags:
+          testing: test
+          delete: on-exit
+      register: output
+    - name: Assert the hsm key created
+      ansible.builtin.assert:
+        that: output.changed
+  rescue:
+    - name: Delete the hsm key
+      azure_rm_keyvaultkey:
+        keyvault_uri: https://hsm{{ rpfx }}.managedhsm.azure.net
+        state: absent
+        key_name: testkey
+
+- name: Get key current version
+  azure_rm_keyvaultkey_info:
+    vault_uri: https://hsm{{ rpfx }}.managedhsm.azure.net
+    name: testkey
+  register: facts
+
+- name: Assert hsm facts
+  ansible.builtin.assert:
+    that:
+      - facts['keys'] | length == 1
+      - facts['keys'][0]['kid']
+      - facts['keys'][0]['permitted_operations'] | length > 0
+      - facts['keys'][0]['type']
+      - facts['keys'][0]['version']
+
+- name: Delete a hsm key
+  azure_rm_keyvaultkey:
+    keyvault_uri: https://hsm{{ rpfx }}.managedhsm.azure.net
+    state: absent
+    key_name: testkey
+  register: output
+
+- name: Assert the hsm deleted
+  ansible.builtin.assert:
+    that: output.changed
+
+#
+# azure_rm_hsmsecret tests
+#
+- name: Create a hsm secret
```

Comment:
Key Vault HSM does not support secret management, this test case can be  removed